client/js/libs/handlebars/parse.js

```diff
@@ -114,7 +115,7 @@ module.exports = function parse(text, users) {
 			const escapedChannel = Handlebars.Utils.escapeExpression(textPart.channel);
 			return `<span class="inline-channel" role="button" tabindex="0" data-chan="${escapedChannel}">${fragments}</span>`;
 		} else if (textPart.emoji) {
-			return `<span class="emoji">${fragments}</span>`;
+			return `<span class="emoji" role="img" aria-label="Emoji: ${emojiMap[textPart.emoji]}" title="${emojiMap[textPart.emoji]}">${fragments}</span>`;
```

Comment:
I tried switching to a tooltip (then remember I already tried that) but our tooltips don't work with `overflow: hidden|scroll`, that's unfortunate. I'd love to solve this (because that's not the only situation, but obviously not in this PR.

Reply:
This still needs an undefiend check. Emojis like 👴🏿 are not in the map (skin color modifier).

Reply:
<img width="351" alt="screen shot 2018-03-18 at 03 23 09" src="https://user-images.githubusercontent.com/113730/37563646-cf44890c-2a5b-11e8-83ac-31bb7f382585.png">

@xPaw any suggestion on what you want to happen? Exclude aria-label/title entirely?

Reply:
It should exclude if it's undefined, yeah.

We could try to work backwards to find an unmodified emoji (e.g. split by ZWJ, or remove skin color modifiers).

Reply:
Probably a future PR though, no? That seems slightly more involved and this is already an improvement. Or add commits to this PR? 😬

Reply:
This PR should at least add a check for undefined.

Reply:
> I tried switching to a tooltip

Just a quick point that I'm not sure we want to use tooltips ever, because I feel like that would be too much. But I'm open to trying it at some point, but probably not.

Reply:
What do you mean too much? That would be consistent with other things like timestamp in terms of styling, and I believe that would allow displaying the value on mobile, which I don't think is possible with `title` (at least not on iOS).
I don't know if that's what you mean, but tooltips don't have to open right away, there is a delay possible like we did on timestamp, to make it unintrusive.

Reply:
As I say, I'm happy to see what it's like but I feel like it would be too much having a tooltip pop up whenever you scroll over an emoji like that, unless it has a longer timeout than the other ones we currently have.